src/ConnectedMode/ConnectedModePackage.cs

```diff
@@ -67,9 +69,11 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             boundSolutionUpdateHandler = componentModel.GetService<BoundSolutionUpdateHandler>();
             timedUpdateHandler = componentModel.GetService<TimedUpdateHandler>();
             localSuppressionsChangedHandler = componentModel.GetService<LocalSuppressionsChangedHandler>();
-            
+
             importBeforeInstallTrigger = componentModel.GetService<ImportBeforeInstallTrigger>();
 
+            hotspotDocumentClosedHandler = componentModel.GetService<IHotspotDocumentClosedHandler>();
```

Comment:
Create an instance of the doc closing handler when the ConnectedMode package loads.

Note: there _might_ be a corner case where this is too late e.g. the user starts VS by opening a bound solution that has an open file that contains hotspots. In that case, there will be a delay before VS loads the package in the background, and the user might close the file in that gap.
I haven't been able to test whether this corner case exists yet as we don't have a working end-to-end. I'll add a note to the Testing ticket.

I tried importing the class in different places e.g. in the `LocalHotspotStore` itself (since we want the lifetime of the doc manager to be the same). However, that creates MEF circular references. I've gone with this solution for now as we need to move forward.